key/src/test/java/software/amazon/kms/key/UpdateHandlerTest.java

```diff
@@ -119,6 +121,7 @@ public void handleRequest_SimpleSuccess() {
             .updateKeyPolicy(eq(proxy), eq(proxyKmsClient), eq(KEY_MODEL_PREVIOUS), eq(KEY_MODEL),
                 eq(callbackContext)))
             .thenReturn(inProgressEvent);
+        // Remove kms:TagResource permission. This throws CfnAccessDeniedException
```

Comment:
There's no CfnAccessDeniedException being thrown below.

I don't think testing softfail logic should fall under the "SimpleSuccess" test case. You should have a test case specific to softfail success